src/sdl_window.cpp

```diff
@@ -19,7 +29,414 @@
+#F8  : reparse keyboard input(this)
+
+#This is a mapping for Bloodborne, inspired by other Souls titles on PC.
+
+#Specifies which joystick the mouse movement controls.
+mouse_to_joystick = right;
+
+#Use healing item, change status in inventory
+triangle = f;
+#Dodge, back in inventory
+circle = space;
+#Interact, select item in inventory
+cross = e;
+#Use quick item, remove item in inventory
+square = r;
+
+#Emergency extra bullets
+up = w, lalt;
+up = mousewheelup;
+#Change quick item
+down = s, lalt;
+down = mousewheeldown;
+#Change weapon in left hand
+left = a, lalt;
+left = mousewheelleft;
+#Change weapon in right hand
+right = d, lalt;
+right = mousewheelright;
+#Change into 'inventory mode', so you don't have to hold lalt every time you go into menus
+modkey_toggle = i, lalt;
+
+#Menu
+options = escape;
+#Gestures
+touchpad = g;
+
+#Transform
+l1 = rightbutton, lshift;
+#Shoot
+r1 = leftbutton;
+#Light attack
+l2 = rightbutton;
+#Heavy attack
+r2 = leftbutton, lshift;
+#Does nothing
+l3 = x;
+#Center cam, lock on
+r3 = q;
+r3 = middlebutton;
+
+#Axis mappings
+#Move
+axis_left_x_minus = a;
+axis_left_x_plus = d;
+axis_left_y_minus = w;
+axis_left_y_plus = s;
+#Change to 'walk mode' by holding the following key:
+leftjoystick_halfmode = lctrl;
+)";
+    return default_config;
+}
+
+// Button map: maps key+modifier to controller button
+std::map<KeyBinding, u32> button_map = {};
+std::map<KeyBinding, AxisMapping> axis_map = {};
+std::map<SDL_Keycode, std::pair<SDL_Keymod, bool>> key_to_modkey_toggle_map = {};
+
+// Flags and values for varying purposes
+int mouse_joystick_binding = 0;
+float mouse_deadzone_offset = 0.5, mouse_speed = 1, mouse_speed_offset = 0.125;
+Uint32 mouse_polling_id = 0;
+bool mouse_enabled = false, leftjoystick_halfmode = false, rightjoystick_halfmode = false;
+
+// A vector to store delayed actions by event ID
+std::vector<DelayedAction> delayedActions;
+
+KeyBinding::KeyBinding(const SDL_Event* event) {
+    modifier = getCustomModState();
+    key = 0;
+    if (event->type == SDL_EVENT_KEY_DOWN || event->type == SDL_EVENT_KEY_UP) {
+        key = event->key.key;
+    } else if (event->type == SDL_EVENT_MOUSE_BUTTON_DOWN ||
+               event->type == SDL_EVENT_MOUSE_BUTTON_UP) {
+        key = event->button.button;
+    } else if (event->type == SDL_EVENT_MOUSE_WHEEL) {
+        key = getMouseWheelEvent(event);
+    } else {
+        std::cout << "We don't support this event type!\n";
+    }
+}
+
+bool KeyBinding::operator<(const KeyBinding& other) const {
+    return std::tie(key, modifier) < std::tie(other.key, other.modifier);
+}
+
+SDL_Keymod KeyBinding::getCustomModState() {
+    SDL_Keymod state = SDL_GetModState();
+    for (auto mod_flag : KBMConfig::key_to_modkey_toggle_map) {
+        if (mod_flag.second.second) {
+            state |= mod_flag.second.first;
+        }
+    }
+    return state;
+}
+
+void parseInputConfig(const std::string game_id = "") {
```

Comment:
Ditto
```suggestion
void parseInputConfig(std::string_view game_id = "") {
```

Reply:
I left this as is, because it would have to be converted to std::string anyway